### PR TITLE
chore(Postfix): disable DNSBLs

### LINKED ISSUE #2976: Double zen.spamhaus.org check is excess?
# Subject
I would like some feedback concerning a use case

## Description
I guess there is unnecessary double check with zen.spamhaus.org in current default DMS configuration:

1. There is a good smart spam scoring check in postscreen (with whitelist scores combination using **list.dnswl.org**)
```
postscreen_dnsbl_sites = zen.spamhaus.org=127.0.0.[2..11]*3 bl.mailspike.net=127.0.0.[2;14;13;12;11;10] b.barracudacentral.org*2 bl.spameatingmonkey.net=127.0.0.2 dnsbl.sorbs.net psbl.surriel.com list.dnswl.org=127.0.[0..255].0*-2 list.dnswl.org=127.0.[0..255].1*-3 list.dnswl.org=127.0.[0..255].[2..3]*-4
```

2. The second check by postfix itself with **smtpd_recipient_restrictions** parameter, where only(!) **zen.spamhaus.org** is checked
```
smtpd_recipient_restrictions = check_recipient_access texthash:/tmp/docker-mailserver/postfix-receive-access.cf, permit_sasl_authenticated, permit_mynetworks, reject_unauth_destination, check_policy_service unix:private/policyd-spf, reject_unauth_pipelining, reject_invalid_helo_hostname, reject_non_fqdn_helo_hostname, reject_unknown_recipient_domain, check_policy_service inet:localhost:65265, reject_rbl_client zen.spamhaus.org=127.0.0.[2..11]
```

So, I think this **reject_rbl_client zen.spamhaus.org=127.0.0.[2..11]** opption for **smtpd_recipient_restrictions** parameter is unnecessary and excess.

Real example:
1. Common SPAM email successfuly blocked by postscreen:
```
Jan  3 13:35:37 persistent postfix/postscreen[1868]: CONNECT from [163.123.140.73]:56883 to [192.168.166.153]:25
Jan  3 13:35:37 persistent postfix/dnsblog[1510659]: addr 163.123.140.73 listed by domain zen.spamhaus.org as 127.0.0.3
Jan  3 13:35:37 persistent postfix/dnsblog[1510655]: addr 163.123.140.73 listed by domain b.barracudacentral.org as 127.0.0.2
Jan  3 13:35:37 persistent postfix/dnsblog[1510660]: addr 163.123.140.73 listed by domain dnsbl.sorbs.net as 127.0.0.6
Jan  3 13:35:37 persistent postfix/dnsblog[1510659]: addr 163.123.140.73 listed by domain psbl.surriel.com as 127.0.0.2
Jan  3 13:35:43 persistent postfix/postscreen[1868]: DNSBL rank 7 for [163.123.140.73]:56883
Jan  3 13:35:44 persistent postfix/postscreen[1868]: NOQUEUE: reject: RCPT from [163.123.140.73]:56883: 550 5.7.1 Service unavailable; client [163.123.140.73] blocked using zen.s
pamhaus.org; from=<59954-94134-7606-13991-***@mail.solarcostmy.rest>, to=<***>, proto=ESMTP, helo=<corona.solarcostmy.rest>
Jan  3 13:35:44 persistent postfix/postscreen[1868]: DISCONNECT [163.123.140.73]:56883
```

2. Good email is passed by postscreen (taking account witn **list.dnswl.org**) but rejected by postfix using **smtpd_recipient_restrictions** later:
```
Jan  3 19:56:55 persistent postfix/postscreen[1868]: CONNECT from [192.30.252.201]:36709 to [192.168.166.153]:25
Jan  3 19:56:55 persistent postfix/dnsblog[1566326]: addr 192.30.252.201 listed by domain list.dnswl.org as 127.0.9.3
Jan  3 19:56:55 persistent postfix/dnsblog[1566327]: addr 192.30.252.201 listed by domain zen.spamhaus.org as 127.0.0.3
Jan  3 19:56:55 persistent postfix/postscreen[1868]: PASS NEW [192.30.252.201]:36709
Jan  3 19:56:55 persistent postfix/smtpd[1566330]: connect from out-18.smtp.github.com[192.30.252.201]
Jan  3 19:56:56 persistent postfix/smtpd[1566330]: Anonymous TLS connection established from out-18.smtp.github.com[192.30.252.201]: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
Jan  3 19:56:56 persistent policyd-spf[1566337]: prepend Received-SPF: Pass (mailfrom) identity=mailfrom; client-ip=192.30.252.201; helo=smtp.github.com; envelope-from=noreply@github.com; receiver=<UNKNOWN>
Jan  3 19:56:56 persistent postfix/smtpd[1566330]: NOQUEUE: reject: RCPT from out-18.smtp.github.com[192.30.252.201]: 554 5.7.1 Service unavailable; Client host [192.30.252.201] blocked using zen.spamhaus.org; https://www.spamhaus.org/sbl/query/SBLCSS; from=<noreply@github.com> to=<***> proto=ESMTP helo=<smtp.github.com>
Jan  3 19:56:56 persistent postfix/smtpd[1566330]: disconnect from out-18.smtp.github.com[192.30.252.201] ehlo=2 starttls=1 mail=1 rcpt=0/1 data=0/1 rset=1 quit=1 commands=6/8
```

This is for version DMS 11.3.0 (linux, arm, k8s) with default config files.

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

_Postfix_ should not handle DNSBLs anymore. We should let _Postscreen_ handle this.

Closes #2976

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by polarathene** [comment]
> _Postfix_ should not handle DNSBLs anymore. We should let _Postscreen_ handle this.

I went over #2976 and looked up some other resources and agree :+1: 

---

This [Spamhaus Postfix docs](https://docs.spamhaus.com/datasets/docs/source/40-real-world-usage/PublicMirrors/MTAs/020-Postfix.html) advises to defer to SpamAssassin or Rspamd via milter if either of those are setup, otherwise to prefer Postscreen. **Should Postscreen still be enabled if Rspamd has the RBL module active?**

This [old Mailing list discussion with Postfix maintainers responding](https://mailing.postfix.users.narkive.com/xTDM4a3w/postscreen-dnsbl-sites-vs-reject-rbl-client) about `reject_rbl_client` vs `postscreen_dnsbl_sites`:

<details>
<summary>Quotes from mailing list discussion</summary>

> On the interfaces and ports that postscreen(8) passes mail to, yes.
> If you have a dedicated submission port, this is not affected by postscreen running on port 25.

> The other difference is that postscreen caches a "pass" dnsbl result for `postscreen_dnsbl_ttl` (default 1h).
> Some sites may prefer to lower the cache TTL or do the tests in smtpd to quickly catch previously good clients gone bad, or to increase the TTL to reduce DNS lookups and latency.
> http://www.postfix.org/postconf.5.html#postscreen_dnsbl_ttl

> Note that postscreen caches the results of successful tests, so that it does not repeat every test for every connection. > This is controlled by the `postscreen_mumble_ttl` parameters.

> - postscreen caches that the client IS NOT listed in DNSBL. It doesn't cache clients that are listed.
> - DNS servers cache that the client IS listed in DNSBL. They don't cache non-existent DNSBL records.

> I can give an example of when/why they might help.
> Under stress, postscreen reduces the greet pause to 2 seconds. Under stress, the possibility that DNSBL responses might be delayed is greater.
> Why would you not avail yourself of that second chance to query `zen.spamhaus.org`? It's cached now at your nameserver, whether positive or negative, so it hurts nothing.

</details>

- The last quote cites one reason you may want both, but as https://github.com/docker-mailserver/docker-mailserver/issues/2976#issuecomment-1375620232 pointed out, that can likewise cause unwanted rejection from Postfix that a more advanced Postscreen check would have permitted.
- Postscreen typically only applies to inbound mail on port 25. Not likely a concern and a user can configure that if they need to? (_besides [postscreen shouldn't be used on submission ports](https://github.com/docker-mailserver/docker-mailserver/issues/838#issuecomment-365520291)_)
- Overall favors preferring postscreen only.

---

### Related issues / PRs for improved traceability

Mostly for my benefit :sweat_smile: 

- Dec 2022 [feature request for Spamhaus DQS (Data Query Service) support](https://github.com/docker-mailserver/docker-mailserver/issues/2955). User appears to be manually approaching it with `user-patches.sh` and this may be a breaking change for them?
- Dec 2022 [issue with change in Spamhaus policies with queries not configured for DQS](https://github.com/docker-mailserver/docker-mailserver/issues/2918#issuecomment-1335287973) (_[associated discussion thread](https://github.com/docker-mailserver/docker-mailserver/discussions/2917)_), also references issues with SpamAssassin and Postscreen (resolved by @georglauterbach ). Also details benefit of approach with running your own recursive DNS resolver.
- Nov 2022 [issue about Spamhaus changes and error code handling](https://github.com/docker-mailserver/docker-mailserver/issues/2889#issuecomment-1303834324), resolved by [this PR](https://github.com/docker-mailserver/docker-mailserver/pull/2890).
- July 2022 [issue about Spamhaus causing blocks for all mail, despite not being listed](https://github.com/docker-mailserver/docker-mailserver/issues/2704#issuecomment-1200070216), which is likely explained by issues referenced above. `postscreen` was still causing blocks due to the spamhaus response though, not only Postfix `reject_rbl_client`. They also state `ENABLE_DNSBL=0` didn't help though :man_shrugging: 
- Jan 2022 [introduced `ENABLE_DNSBL`](https://github.com/docker-mailserver/docker-mailserver/pull/2342). Releases with this feature onwards opt-out of DNSBLs.
- Feb 2021 [issue about `bl.spamcop.net`](https://github.com/docker-mailserver/docker-mailserver/issues/1781) failing due to expiry. Noted as any lookup to that blocklist was considered a failure preventing legitimate mail from being received. Removed by https://github.com/docker-mailserver/docker-mailserver/pull/1782
- April 2020 [issue about internal only setup being slowed down by DNSBLs that intentionally were unreachable](https://github.com/docker-mailserver/docker-mailserver/issues/1456), since fixed by opt-out with `ENABLE_DNSBL=0`.
- Jan 2018 [introduced postscreen and `postscreen_dnsbl_sites` configured blocklists](https://github.com/docker-mailserver/docker-mailserver/pull/799). Briefly asks about dropping Postgrey since Postscreen can replace it, but in the [associated issue](https://github.com/docker-mailserver/docker-mailserver/issues/780#issuecomment-361170045) favors keeping Postgrey for more granular support. Both PR and issue quote from [Postfix postscreen docs](https://www.postfix.org/POSTSCREEN_README.html), such as the 4 layers of defense with Postscreen at the first layer, Postfix for the next two, and third-party service for the fourth on content (Amavis/SA, rspamd).
- April 2016 [`smtpd_recipient_restrictions` updated to include three `reject_rbl_client` blocklists](https://github.com/docker-mailserver/docker-mailserver/pull/165), one of these was [removed not long after in May 2016](https://github.com/docker-mailserver/docker-mailserver/issues/197#issuecomment-221899879) where it was reported as often blocking mail from popular vendors such as Gmail (_the [referenced wikipedia article also cites this](https://en.wikipedia.org/wiki/Spam_and_Open_Relay_Blocking_System)_), while the newer postscreen list has brought back that SORBS blocklist :sweat_smile: (_**NOTE:** Possibly useful reference of DNSBLs](https://en.wikipedia.org/wiki/Comparison_of_DNS_blacklists), one of these is [Abusix, which DMS received a recent Feature Request to support](https://github.com/docker-mailserver/docker-mailserver/issues/3003)_)

### DNS

Some advice over the years with the project for working around the many reports of false-positive blockings from the DNSBLs with public DNS resolvers:

- April 2018 - Googles `8.8.8.8` won't work, but Cloudflares `1.1.1.1` would: https://github.com/docker-mailserver/docker-mailserver/issues/948#issuecomment-380717902
- June 2020 - Then neither of those but Quad9 `9.9.9.9` would: https://github.com/docker-mailserver/docker-mailserver/issues/1542#issuecomment-647092816
- Jan 2021 - Use them all? :man_shrugging: : https://github.com/docker-mailserver/docker-mailserver/issues/1742#issuecomment-756074164
- Nov 2021 - None will work, but `9.9.9.11` will because of ECS feature: https://github.com/docker-mailserver/docker-mailserver/issues/1927#issuecomment-955713448
   
   > The easiest solution would be to add a dns server which supports ECS.
   > Spamhaus changed their policy in 2020 of accepting any query towards denying queries send via open resolvers due to misuse of their service.
   > If you still identify yourself with your own IP they are still happy to serve you. And you'll keep their pretty good filter working.
   > 
   > ```yaml
   >   dns:
   >     - 9.9.9.11 #QUAD9 DNS Server with ECS enabled see: https://www.quad9.net/service/service-addresses-and-features/#ecssec
   >   ```
- July 2022 - None works properly, DIY: https://github.com/docker-mailserver/docker-mailserver/issues/2697#issuecomment-1195805186
   
   > Spam filtering tools like SA, but also Postfix/Postscreen using DNSBLs, _require_ a **non-official**, **recursive** DNS resolver.
   > If you try to use official DNS servers (`1.1.1.1`, `9.9.9.9`, etc.), you'll find that the answers to your queries are:
   > 
   > 1. at best, meaningless
   > 2. at worst, classifying legit mails as spam if Postfix for example is not configured correctly
   > 
   > So, I figured: why not resolve the traffic to SpamHaus, Sorbs, DNSWL, SpamEatingMonkey etc. on my own and use official DNS servers for all the other DNS traffic.

**Review by polarathene — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**
